web/ui/react-app/src/pages/targets/ScrapePoolSelector.tsx

```diff
@@ -0,0 +1,202 @@
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const toggle = () => setDropdownOpen((prevState) => !prevState);
+
+  const [filter, setFilter] = useState<string>('');
+
+  return (
+    <Dropdown isOpen={dropdownOpen} toggle={toggle}>
+      <DropdownToggle caret className="mw-100 text-truncate">
+        {initialValue ? initialValue : 'Select scrape pool'}
+      </DropdownToggle>
+      <DropdownMenu style={{ maxHeight: 400, overflowY: 'auto' }}>
+        <DropdownItem key="__header" header toggle={false}>
+          <Input autoFocus placeholder="Filter" value={filter} onChange={(event) => setFilter(event.target.value.trim())} />
+        </DropdownItem>
+        {scrapePools.length === 0 ? (
+          <DropdownItem disabled>No scrape pools configured</DropdownItem>
+        ) : (
+          scrapePools
+            .filter((name) => filter === '' || name.includes(filter))
+            .map((name) => (
+              <DropdownItem key={name} value={name} onClick={() => onScrapePoolChange(name)}>
+                {name}
+              </DropdownItem>
+            ))
+        )}
+      </DropdownMenu>
+    </Dropdown>
+  );
+};
+
+const ScrapePoolSelector: FC<ScrapePoolNamesListProps> = ({ scrapePools }) => {
+  // get initially selected scrape pool name from query params
+  let initialValue: string | null = getScrapePoolNameFromQueryParams();
+  // but validate that passed name is one of existing scrape pools we got from the API
+  if (initialValue !== null && !scrapePools.includes(initialValue)) {
+    // name from query params is not valid, reset initial name to null
+    initialValue = null;
+  }
+  const [scrapePool, setScrapePool] = useState<string | null>(initialValue);
+
+  const onNameChange = useCallback((name: string) => {
+    setScrapePool(name);
+    setScrapePoolNameQueryParams(name);
+  }, []);
+
+  const [filters, setFilters] = useLocalStorage('target-health-filter', {
+    healthy: true,
+    unhealthy: true,
+    unknown: true,
+  });
+  const toggleFilter = (val: targetHealth) => () => {
+    setFilters({
+      ...filters,
+      [val]: !filters[val],
+    });
+  };
+
+  const [search, setSearch] = useState(getQuerySearchFilter());
+  const handleSearchChange = useCallback(
+    (value: string) => {
+      setQuerySearchFilter(value);
+      setSearch(value);
+    },
+    [setSearch]
+  );
+
+  return (
+    <>
+      <Row className="align-items-center my-2">
+        <Col lg="4" md="5" className="py-1">
+          <ScrapePoolDropDown initialValue={initialValue} scrapePools={scrapePools} onScrapePoolChange={onNameChange} />
+        </Col>
+        <Col lg="5" md="4" className="py-1">
+          <SearchBar
+            defaultValue={search}
+            handleChange={(v) => handleSearchChange(v.trim())}
+            placeholder="Filter by endpoint or labels"
+          />
+        </Col>
+        <Col className="d-flex flex-row-reverse py-1" md="3">
+          {healthColorTuples.map(([val, color]) => (
```

Comment:
Nice, I actually like these filters better than the buttons we have currently (esp. since the buttons are grouped together with the unrelated collapse/expand button). But I still think we should keep these filters looking the same even in the multi-pool view. So something like:

![filters](https://user-images.githubusercontent.com/538008/188477231-c0a41226-981c-4a90-989c-aeea03e026fe.png)